ledger/accountdb_test.go

```diff
@@ -1078,6 +1080,81 @@ func BenchmarkWriteCatchpointStagingBalances(b *testing.B) {
 	}
 }
 
+func TestKeyPrefixIntervalPreprocessing(t *testing.T) {
+	testCases := []struct {
+		input            []byte
+		outputPrefix     []byte
+		outputPrefixIncr []byte
+	}{
+		{input: []byte{0xAB, 0xCD}, outputPrefix: []byte{0xAB, 0xCD}, outputPrefixIncr: []byte{0xAB, 0xCE}},
+		{input: []byte{0xFF}, outputPrefix: []byte{0xFF}, outputPrefixIncr: nil},
+		{input: []byte{0xFE, 0xFF}, outputPrefix: []byte{0xFE, 0xFF}, outputPrefixIncr: []byte{0xFF, 0x00}},
```

Comment:
Shouldn't we require `outputPrefixIncr` of `[]byte{0xFE, 0xFF}` to be `[]byte{0xFF}` ?
If not, we have an unintended solution `x = []byte{0xFF}` to the constraints:
```
[]byte{0xFE, 0xFF} <= x < []byte{0xFF, 0x00}
```

Reply:
good find, I think this argument makes sense. I will put this as part of the test case in DB test, rather than just a unit test.